charts/jenkins/templates/jenkins-controller-ingress.yaml

```diff
@@ -1,11 +1,11 @@
 {{- $kubeTargetVersion := default .Capabilities.KubeVersion.GitVersion .Values.kubeTargetVersionOverride }}
 {{- if .Values.controller.ingress.enabled }}
-{{- if semverCompare ">=1.19-0" $kubeTargetVersion -}}
+{{- if .Values.controller.ingress.apiVersion -}}
+apiVersion: {{ .Values.controller.ingress.apiVersion }}
+{{- else if semverCompare ">=1.19-0" $kubeTargetVersion -}}
```

Comment:
This change will alter existing behavior, because `apiVersion` is set to `"extensions/v1beta1"` in values.yaml https://github.com/jenkinsci/helm-charts/blob/7273190/charts/jenkins/values.yaml#L424

So all users would get that apiVersion.

Reply:
We could comment that out? Any suggestions as to how to get around this then? If we are setting the ingress.apiVersion in the values, it makes sense to me that we use that version. Globally setting the API version to something it isn't to to enable a variable for a specific resource seems like an anti-pattern. `kubeTargetVersionOverride` isn't defined in the values.yaml, so I feel like this is asking a lot from users who don't know Helm or how to manually read templates to get an ingress to work.

Reply:
Just have no value in values.yaml
```
   apiVersion:
```

The benefit over commenting it out completely is that you are able to get auto completion in some IDEs.